subsys/bluetooth/host/l2cap.c

```diff
@@ -895,6 +895,8 @@ static struct net_buf *l2cap_chan_le_get_tx_buf(struct bt_l2cap_le_chan *ch)
 static int l2cap_chan_le_send_sdu(struct bt_l2cap_le_chan *ch,
 				  struct net_buf *buf);
 
+static inline bt_l2cap_chan_state_t bt_l2cap_chan_get_state(struct bt_l2cap_chan *chan);
```

Comment:
Is there a reason why you con't just rearrange the functions (i.e. move the implementation of this one higher up in the file) rather than adding a forward declaration? It's usually only in circular dependency situations where forward declarations of static functions are used.

Reply:
can't win lol. I'm sure if I did that some people would've complained that diff is unnecessarily large :|

Reply:
Yeah, I was just about to say that I usually prefer this approach, as `git` would take it as you've modified all those lines which kind of messes up the git history :D

Reply:
I didn't mean to imply that you should do that in the same commit. If you did that then it does indeed mess up the diff. Normally reorganizing is done as a separate prior commit, with a commit message that explains why it's done.

Reply:
> Yeah, I was just about to say that I usually prefer this approach, as `git` would take it as you've modified all those lines which kind of messes up the git history :D

If you're talking about `git blame` then it has switches that can be used to avoid this kind of false-positives. I learned a bunch of these from the git talk that was presented at this year's FOSDEM:
https://gist.github.com/Drarig29/5580ccd123988af4fc6da1a2cacca054